Adapted to xenko 3 and some improvements

tebjan opened the pull request.

stefnotch committed:
Some fixes

stefnotch committed:
Er, that was some debugging code

stefnotch committed:
Merge remote-tracking branch 'SvenEV/master'

tebjan committed:
updated to latest xenko distribution and updated nuget references

tebjan committed:
fixed shader declaration parsing and added class to syntax highlighting

tebjan committed:
upped version to 1.1

tebjan committed:
showing version in info window

tebjan committed:
also showing version in main window

SvenEV commented on XenkoShaderExplorer/MainViewModel.cs:
```diff
@@ -66,13 +66,14 @@ public MainViewModel()
             {
-                var xenkoDir = Environment.GetEnvironmentVariable(XenkoEnvironmentVariable) ?? FallbackBasePath;
-                var basePath = System.IO.Path.Combine(xenkoDir, "GamePackages");
+                var userDir = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
```
You fixed the folder lookup here which is nice 👍 However, I would like to keep the ability to define the Xenko folder via the "XenkoDir" environment variable - this way you can use the Shader Explorer with a development clone of the Xenko repository, not just with the official releases. So maybe you could first check if "XenkoDir" is set and if so, use that directory instead of the automatically detected one.

tebjan replied:
yes, agreed. and maybe also an additional folder, like your current project, so shaders from that also get added to the tree view. i'll have a look at it.

SvenEV commented on XenkoShaderExplorer/MainViewModel.cs:
```diff
@@ -143,26 +144,27 @@ private IEnumerable<Shader> BuildShaderTree()
-                            shader.BaseShaders.Add(baseShader);
-                            baseShader.DerivedShaders.Add(shader);
+                        System.Diagnostics.Debug.WriteLine(string.Join(", ", baseShaderNames));
+                        //if (!baseShaderNames.Contains("ShadowMapCasterBase"))
```
Please remove the `Debug.WriteLine`-call and outdated comments

SvenEV commented on Stride.ShaderExplorer/InfoWindow.xaml:
```diff
@@ -9,12 +9,11 @@
             <StackPanel>
 
-                <TextBlock Text="Shader Explorer for Xenko"
+                <TextBlock Name="InfoHeaderBlock" Text="{Binding InfoHeader}"
```
The `Text` binding can be removed I think

SvenEV commented on XenkoShaderExplorer/MainViewModel.cs:
```diff
@@ -126,8 +127,8 @@ private IEnumerable<Shader> BuildShaderTree()
-                var declaration = string.Join(" ", File.ReadAllLines(shader.Path)
-                    .SkipWhile(s => !s.Trim().StartsWith("shader")) // From "shader"
+                var declaration = string.Join(" ", File.ReadLines(shader.Path)
+                    .SkipWhile(s => !s.Trim().Contains("shader ") && !s.Trim().Contains("class ")) // From "shader" or "class"
```
You replaced `StartsWith` with `Contains` here which seems to break parsing of some shaders. For example, consider "SubsurfaceScatteringBlurShader.xksl": It starts with a long comment block containing the line "It's a simple **shader** composed of two components". By using `Contains`, we incorrectly start parsing the shader declaration at that comment line, resulting in the shader not being listed at all in the tree view.

Do you have specific examples where the old `StartsWith`-check is not sufficient?

tebjan replied:
yes there were some shaders that had some keywords before the `shader` declaration. hm, interesting one... the parsing might have to get smarter. i'll think about it. thanks for the review!

SvenEV reviewed (changes requested):
Thank you for your help, tebjan! I have tried your Shader Explorer version and it worked with my local Xenko installation, that's nice. A few things need to be addressed, though, before I can merge this (see comments). Let me know if you need any clarification or help.

tebjan committed:
added solection for official and dev folders. adapted code for new xenko nuget layout.

tebjan commented:
i've done some improvements regarding the new package layout but haven't looked at the parsing issue yet.

profan committed:
allow it to work when NUGET_PACKAGES is set explicitly.

as you can set NUGET_PACKAGES globally to specify where packages end up,
this change is required for it to look in the right place when the env var is
set.

tebjan committed:
Merge pull request #1 from profan/master

allow it to work when NUGET_PACKAGES is set explicitly.

tebjan committed:
incremented version to 1.1.2.0

tebjan committed:
fixed nuget directory filter

tebjan committed:
Update README.md

tebjan committed:
Updated and renamed to Stride

tebjan committed:
Update README.md

tebjan committed:
Incemented version to 1.2.0.0

tebjan committed:
Merge branch 'master' of https://github.com/tebjan/Stride.ShaderExplorer

tebjan committed:
Catching empty dir cases

tebjan committed:
Update README.md

tebjan committed:
Improves SDSL syntax highlighting with more keywords

tebjan committed:
Converts projects to SDK style, references Stride and adds shader parser files from VL.Stride

tebjan committed:
Adds AdditionalPaths and saves it in user settings

tebjan committed:
Shaders are now parsed by the stride shader parser

tebjan committed:
Changes font to Consolas and code auto formatting

tebjan committed:
Adds word selection to code box with info bar

tebjan committed:
Adds base shader drop down

tebjan committed:
Adds navigation and WIP for var and method name search

tebjan committed:
Adds shader member finder and uses list boxes for base, derived and member shaders

tebjan committed:
Changed theme to dark

tebjan committed:
UI clenaup and styling

tebjan committed:
Displays overloads as list of members, adds screenshot

tebjan committed:
Updates README.md

tebjan committed:
Changes version to 2.0.0

tebjan committed:
Update README.md

tebjan committed:
Updates to .net 8 and auto finds vvvv paths and nugets.

tebjan committed:
Adds path and nuget resolvers

tebjan committed:
Adds member search and better architecture.

tebjan committed:
Enhance README with vvvv gamma and search features

Updated README to include vvvv gamma in the tool description and enhanced search functionality.

tebjan committed:
vs code extension initial setup

tebjan committed:
Adds logo and a few features

tebjan committed:
Helpers for some tasks

tebjan committed:
Add automated VS Code extension publishing via GitHub Actions

- Add release workflow triggered by vscode-v* tags
- Configure .NET Install Tool dependency for runtime acquisition
- Add sidebar panels for inheritance, streams, variables, methods
- Support pre-release publishing with hyphenated version tags

tebjan committed:
Add automated VS Code extension publishing via GitHub Actions

- Add release workflow triggered by vscode-v* tags
- Configure .NET Install Tool dependency for runtime acquisition
- Add sidebar panels for inheritance, streams, variables, methods
- Support pre-release publishing with hyphenated version tags

tebjan committed:
Better tree menu and clickable base shaders and types

tebjan committed:
Some parsing improvements and completion sorting

tebjan committed:
Add templates detection and better tree state handling and easy jump to shader

tebjan committed:
linter error reports and cleaner local shader members

tebjan committed:
Change license from MIT to GPL-3.0

tebjan committed:
add extension to readme
